src/screens/home.tsx

```diff
@@ -1,17 +1,125 @@
-import { StyleSheet, View } from 'react-native'
+import { useState } from 'react'
+import { StyleSheet, View, Linking, FlatList } from 'react-native'
 import { Button, Text } from 'react-native-paper'
+import * as repl from 'repl'
+import { config } from '../config'
 import { ScreenNavigationProps } from '../routes'
+import SelectDropdown from 'react-native-select-dropdown'
 
 type HomeScreenProps = ScreenNavigationProps<'Home'>
 
-const HomeScreen = ({ navigation }: HomeScreenProps) => (
-  <View style={styles.container}>
-    <Text style={styles.text}>Home Screen</Text>
-    <Button mode="contained" onPress={() => navigation.navigate('Details')}>
-      Go to details
-    </Button>
-  </View>
-)
+const tramStops = [
+  "St Peter''s Square",
+  'Chorlton',
+  'Old Trafford',
+  'Cornbrook',
+  'Firswood',
+]
+
+interface TFGMResponse {
+  [index: string]: string
+  Dest0: string
+  Dest1: string
+  Dest2: string
+  Dest3: string
+  Wait0: string
+  Wait1: string
+  Wait2: string
+  Wait3: string
+}
+
+interface TFGMRawResponse {
+  value: TFGMResponse[]
+}
+
+const HomeScreen = ({ navigation }: HomeScreenProps) => {
+  const [departures, setDepartures] = useState<
+    { destination: string; time: string }[]
+  >([])
+  const [station, setStation] = useState('')
+  const liveDepartures: { destination: string; time: string }[] = []
+  const uniqueCheck: string[] = []
+
+  function addUniqueDepartureEntry(jsonObject: TFGMResponse, i: number) {
+    if (jsonObject['Wait' + String(i)]) {
+      const destinationTime = {
+        destination: jsonObject['Dest' + String(i)],
+        time: jsonObject['Wait' + String(i)],
+      }
+      const departureString = String(
+        String(jsonObject['Dest' + String(i)] + jsonObject['Wait' + String(i)])
+      )
+      if (!uniqueCheck.includes(departureString)) {
+        liveDepartures.push(destinationTime)
+        uniqueCheck.push(departureString)
+      }
+    }
+  }
+
+  function extractDepartureFromApiObject({
+    jsonObject,
+  }: {
+    jsonObject: TFGMResponse
+  }) {
+    for (let i = 0; i < 4; i++) {
+      addUniqueDepartureEntry(jsonObject, i)
+    }
+  }
+
+  const fetchDataFromTFGMApi = async (station: string) => {
+    try {
+      const response: Response = await fetch(
+        `https://api.tfgm.com/odata/Metrolinks?$filter=StationLocation eq '${station}'`,
+        {
+          method: 'GET',
+          headers: {
+            'Ocp-Apim-Subscription-Key': config.apiKey,
+          },
+        }
+      )
+      const json = (await response.json()) as TFGMRawResponse
+      for (const obj of json.value) {
+        extractDepartureFromApiObject({ jsonObject: obj })
+      }
+      liveDepartures.sort((a, b) => (Number(a.time) < Number(b.time) ? -1 : 1))
+      setDepartures(liveDepartures)
+      return json
+    } catch (error) {
+      console.error(error)
+    }
+  }
```

Comment:
There's a lot in this component, and you're doing multiple things at once in a way that is difficult to read.

It's generally a good idea to have small chunks of code in functions that self-document and in files that represent the purpose(s) of that file. E.g., we can have a new file `src/clients/metrolinks.ts` which calls the API, formats the response, and returns the formatted data. Then you could call that function from your component and set that to the state.

I think you should have that logic in a different file because the act of calling that endpoint and formatting that data isn't inherently tied to the component itself.